### PR TITLE
pixel perfect canvas rendering of box and block characters

### LINKED ISSUE #2409: Manually draw pixel-perfect glyphs for Box Drawing and Block Elements characters
## Details
- Browser and browser version: Chrome 76.0.3809.132 (64 bit), FireFox 69.0 (64 bit)
- OS version: Windows 7 Pro 64 bit
- xterm.js version: ^3.14.5

### Steps to reproduce

index.html:
```
<!doctype html>
<html>

<head>
	<link rel="stylesheet" href="node_modules/xterm/dist/xterm.css" />
	<script src="node_modules/xterm/dist/xterm.js"></script>
</head>

<body>
	<div id="terminal"></div>
	<script>
		var options = {
			rows: 30,
			cols: 80,
			fontFamily: '"Courier New", "DejaVu Sans Mono", "Everson Mono", FreeMono, "Andale Mono", monospace',
			fontSize: 12,
			convertEol: true
		};
		var term = new Terminal(options);
		term.open(document.getElementById('terminal'));

		term.write('           ╔═════════════════════════════════════════════════════════╕\n');
		term.write('           ║                                                         │\n');
		term.write('           ║              ╔═══════════════╦════════╤════════╗        │\n');
		term.write('           ║              ║               ║        │        ║        │\n');
		term.write('           ║              ║               ║        │        ║        │\n');
	</script>
</body>

</html>
```

image

There are several related issues but it does not help:

https://github.com/xtermjs/xterm.js/issues/475
https://github.com/xtermjs/xterm.js/issues/992

### PR DESCRIPTION
**Pull request by meganrogge**
Fixes #2409

### CODE REVIEW

File: demo/client.ts [hunk 1]
```diff
@@ -214,16 +214,17 @@ function createTerminal(): void {
...
-        socket = new WebSocket(socketURL);
-        socket.onopen = runRealTerminal;
-        socket.onclose = runFakeTerminal;
-        socket.onerror = runFakeTerminal;
-      });
-    });
+    // fetch('/terminals?cols=' + term.cols + '&rows=' + term.rows, {method: 'POST'}).then((res) => {
+    //   res.text().then((processId) => {
+    //     pid = processId;
+    //     socketURL += processId;
```
Comment (meganrogge) [inline demo/client.ts:220]: leaving this bc you might want to fine tune the dashed characters

**Comment by Tyriar** [comment]
Progress (canvas):

image

Old webgl rendering:

image

**Comment by Tyriar** [comment]
w3 alignment tests in canvas:

image

old in webgl:

image

**Comment by Tyriar** [comment]
Webgl line height, letter spacing works. Plus the shade characters work 🎉 

image

**Comment by Tyriar** [comment]
Had a think about testing and can't really think of a nice way to do it across all browsers unless we managed to get it drawing exactly across all browsers. But FF and Chrome seem to differ in how they draw corners of paths so not sure. Going to go ahead on this without tests for now, there is a button at the bottom of the demo that writes the alignment tests to the terminal though.

**Comment by Tyriar** [comment]
Moving ahead with this, here's canvas (1.1 line height, 1 letter spacing):

image

Webgl:

image

**Pull request merged**